src/org/yccheok/jstock/gui/JStock.java

```diff
@@ -470,6 +470,11 @@ public void stateChanged(javax.swing.event.ChangeEvent evt) {
         this.jTable1.getTableHeader().addMouseListener(new TableColumnSelectionPopupListener(1));
         this.jTable1.addMouseListener(new TableRowPopupListener());
         this.jTable1.addKeyListener(new TableKeyEventListener());
+        jTable1.addMouseListener(new java.awt.event.MouseAdapter() {
+            public void mouseClicked(java.awt.event.MouseEvent evt) {
+                jTable1MouseClicked(evt);
+            }
+        });
```

Comment:
I have no problem with that. But that change could impact more then just what I touched if I understand you correctly. For example, the original popup was NOT row specific. Now it follows what I would call more standard window design, first selecting the row that the user right-clicked on. So my point is, that may have other ramifications. I was only focused on the charts. On the plus side, at least the user no longer needs to do 2 mouse actions to get to the chart and indeed has many other alternatives such as ENTER key.